### PR TITLE
b-fix/gh-1148-ignored-exceptions

### PR DESCRIPTION
**Pull request by nefeyel**
Solving #1148

### CODE REVIEW

File: src/main/java/eu/execom/hawaii/service/SendNotificationsService.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import eu.execom.hawaii.dto.NotificationDataDto;
 import eu.execom.hawaii.dto.NotificationDto;
 import eu.execom.hawaii.dto.PushNotificationDto;
+import eu.execom.hawaii.exceptions.*;
```
Comment (nefeyel) [inline src/main/java/eu/execom/hawaii/service/SendNotificationsService.java]: changed

File: src/main/java/eu/execom/hawaii/security/TokenProvider.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package eu.execom.hawaii.security;
 
 import eu.execom.hawaii.configuration.AppProperties;
+import eu.execom.hawaii.exceptions.*;
```
Comment (nefeyel) [inline src/main/java/eu/execom/hawaii/security/TokenProvider.java]: changed

File: src/main/java/eu/execom/hawaii/service/EmailService.java [hunk 3]
```diff
@@ -306,6 +310,7 @@ private void createEmail(Email email, MimeMessageHelper helper, String emailText
       helper.getMimeMessage().setContent(emailText, "text/html;charset=utf-8");
     } catch (MessagingException exception) {
       log.error("Error creating email with details: {}", exception.getMessage());
+      throw new ActionNotAllowedException("Error creating email with details: {}" + exception.getMessage());
```
Comment (stefankaticec) [inline src/main/java/eu/execom/hawaii/service/EmailService.java]: This must not be a conflict.

File: src/main/java/eu/execom/hawaii/service/GoogleCalendarService.java [hunk 4]
```diff
@@ -59,6 +62,7 @@ public void init() {
       googleCredential = GoogleCredential.fromStream(new FileInputStream(serviceAccountCredentials));
     } catch (IOException e) {
       log.error("Error reading credentials from file. Exception raised: {}", e);
+      throw new UnauthorizedAccessException("Error reading credentials from file. Exception raised: {}" + e);
```
Comment (stefankaticec) [inline src/main/java/eu/execom/hawaii/service/GoogleCalendarService.java]: Not unauthorized.

File: src/main/java/eu/execom/hawaii/service/AuditInformationService.java [hunk 5]
```diff
@@ -93,6 +94,8 @@ private String serialize(Audit objectForSerialization) {
       currentValue = OBJECT_MAPPER.writeValueAsString(objectForSerialization);
     } catch (JsonProcessingException e) {
       log.error("Failed to map object of class '{}' as string", objectForSerialization.getClass());
+      throw new ActionNotAllowedException("Failed to map object of class '{}' as string" +
```
[the remark this answers is not in the document]
Comment (stefankaticec) [reply src/main/java/eu/execom/hawaii/service/AuditInformationService.java]: This isn't fixed.

File: src/main/java/eu/execom/hawaii/service/EmailService.java [hunk 6]
```diff
@@ -285,6 +287,7 @@ private void sendEmail(Email email) {
       emailSender.send(message);
     } catch (MailException exception) {
       log.error("Error on sending email to: " + email.getTo(), exception);
+      throw new BadRequestException("Error on sending email to: " + email.getTo() + exception);
```
[the remark this answers is not in the document]
Comment (stefankaticec) [reply src/main/java/eu/execom/hawaii/service/EmailService.java]: This isn't fixed.

File: src/main/java/eu/execom/hawaii/service/EmailService.java [hunk 7]
```diff
@@ -294,6 +297,7 @@ private Template getTemplate(String templateName) {
       template = freemarkerConfig.getTemplate(templateName);
     } catch (IOException exception) {
       log.error("Unable to find template with name: {}", templateName);
+      throw new BadRequestException("Unable to find template with name: {}" + templateName);
```
[the remark this answers is not in the document]
Comment (stefankaticec) [reply src/main/java/eu/execom/hawaii/service/EmailService.java]: This isn't fixed.

File: src/main/java/eu/execom/hawaii/service/EmailService.java (hunk 3 above)
Comment (stefankaticec) [reply src/main/java/eu/execom/hawaii/service/EmailService.java]: This isn't fixed.

File: src/main/java/eu/execom/hawaii/service/GoogleCalendarService.java (hunk 4 above)
Comment (stefankaticec) [reply src/main/java/eu/execom/hawaii/service/GoogleCalendarService.java]: This isn't fixed.

File: src/main/java/eu/execom/hawaii/service/SendNotificationsService.java [hunk 8]
```diff
@@ -152,6 +152,7 @@ private String objectToJsonMapper(Object result) {
       jsonInString = objectMapper.writeValueAsString(result);
     } catch (IOException e) {
       log.error("Failed to send push notification {}", e.getMessage());
+      throw new BadRequestException("Failed to send push notification {}" + e.getMessage());
```
Comment (stefankaticec) [inline src/main/java/eu/execom/hawaii/service/SendNotificationsService.java]: This isn't bad request.

File: src/main/java/eu/execom/hawaii/service/SendNotificationsService.java [hunk 9]
```diff
@@ -180,6 +181,7 @@ private void send(String convertedToJson) {
 
     } catch (IOException e) {
       log.error("Failed to send push notification {}", e.getMessage());
+      throw new BadRequestException("Failed to send push notification {}" + e.getMessage());
```
Comment (stefankaticec) [inline src/main/java/eu/execom/hawaii/service/SendNotificationsService.java]: This isn't bad request.

**Pull request closed without merging**